File: API/Entities/Basket.cs
using System;
using System.Linq;

namespace API.Entities
{
    public class Basket
    {
        public int Id { get; set; }
        public required string BasketId { get; set; }
        public List<BasketItem> Items { get; set; } = [];
        public string? ClientSecret { get; set; }// Chave secreta do cliente para o pagamento
        public string? PaymentIntentId { get; set; } // Identificador do pagamento, usado para atualizar ou verificar o status do pagamento

        public void AddItem(Product product, int quantity)
        {
            if (product == null) throw new ArgumentNullException(nameof(product));
            if (quantity <= 0) throw new ArgumentException("Quantity deve ser maior que zero", nameof(quantity));

            var existingItem = FindItem(product.Id);
            if (existingItem == null)
            {
                Items.Add(new BasketItem
                {
                    ProductId = product.Id,
                    Quantity = quantity,
                    Product = product
                });
            }
            else
            {
                existingItem.Quantity += quantity; // Corrigido: soma apenas a quantidade recebida
            }
        }

        public void RemoveItem(int productId, int quantity)
        {
            if (quantity <= 0) throw new ArgumentException("Quantity deve ser maior que zero", nameof(quantity));
            var item = FindItem(productId);
            if (item == null) return;

            item.Quantity -= quantity;
            if (item.Quantity <= 0) Items.Remove(item);
        }

        private BasketItem? FindItem(int productId)
        {
            return Items.FirstOrDefault(item => item.ProductId == productId);
        }
    }
}

File: client/src/app/models/basket.ts
import { Product } from "./product";

export type Basket =
    {
        basketId: string
        items: Item[]
        clientSecret?: string
        paymentIntentId?: string
        
    }
export class Item 
{
    constructor(product: Product, quantity: number) 
    {
        this.productId = String(product.id);// Convertendo o ID do produto para string
        this.name = product.name;// Atribuindo o nome do produto
        this.price = product.price;// Atribuindo o preço do produto
        this.pictureUrl = product.pictureUrl;// Atribuindo a URL da imagem do produto
        this.brand = product.brand;// Atribuindo a marca do produto
        this.type = product.type;// Atribuindo o tipo do produto
        this.quantity = quantity;// Atribuindo a quantidade do produto
    }
    productId: string;
    name: string;
    price: number;
    pictureUrl: string;
    brand: string;
    type: string;
    quantity: number;

}